go/keymanager/churp/status.go

```diff
@@ -70,13 +56,40 @@ type Status struct {
 	// Policy is a signed SGX access control policy.
 	Policy SignedPolicySGX `json:"policy"`
 
+	// Handoff is the epoch of the last successfully completed handoff.
+	//
+	// The zero value indicates that no handoffs have been completed so far.
+	// Note that the first handoff is special and is called the dealer phase,
+	// in which nodes do not reshare or randomize shares but instead construct
+	// the secret and shares.
+	Handoff beacon.EpochTime `json:"handoff"`
+
+	// The hash of the verification matrix from the last successfully completed
+	// handoff.
+	Checksum *hash.Hash `json:"checksum,omitempty"`
+
 	// Committee is a vector of nodes holding a share of the secret
 	// in the active handoff.
 	//
 	// A client needs to obtain more than a threshold number of key shares
 	// from the nodes in this vector to construct the key.
 	Committee []signature.PublicKey `json:"committee,omitempty"`
 
+	// NextHandoff defines the epoch in which the next handoff will occur.
+	//
+	// If an insufficient number of applications is received, the next handoff
+	// will be delayed by one epoch.
+	NextHandoff beacon.EpochTime `json:"next_handoff"`
+
+	// NextChecksum is the hash of the verification matrix from the current
+	// handoff.
+	//
+	// The first candidate to confirm share reconstruction is the source
+	// of truth for the checksum. All other candidates need to confirm
```

Comment:
> Is there a way to prove this in a way that the consensus layer could easily (cheaply) verify? 

No, at least I don't see it. And taking into account that we can replace checksum with some other commitment.
To prevent DoS attacks, we could add slashing. But 

> Also all other candidates or is a threshold enough?

Threshold is enough. But we want to give participants enough time to do the handoff. So maybe all candidates to complete before the end of an epoch, and threshold at the end of an epoch.

Reply:
Ok, that sounds good. No slashing needed for now.